mesh/mesh.go

```diff
@@ -575,6 +568,15 @@ func (msh *Mesh) persistAggregatedLayerHash(layerID types.LayerID, hash types.Ha
 	}
 }
 
+// GetAggregatedLayerHash returns the aggregated layer hash up to the specified layer
+func (msh *Mesh) GetAggregatedLayerHash(layerID types.LayerID) types.Hash32 {
+	h, err := msh.getAggregatedLayerHash(layerID)
+	if err != nil {
+		return EmptyLayerHash
```

Comment:
shouldn't this return an error? event if the requested layer has no blocks the aggregated hash shouldn't be zero

Reply:
exactly. EmptyLayerHash is actually the way to indicate error for aggregated layer. it will **never** be EmptyLayerHash, even for layer 1.

to be honest i was trying to pass around an error between peers but xdr encoding doesn't work with nil 
`xdr:encodeInterface: can't encode nil interface`